### PR TITLE
common/options.h: Specify limited length of string

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
@liewegas Thanks for dropping `DNM` :)
@jcsp Please check, when you get a chance.

**Comment by jcsp** [comment]
iirc the const char* references went in here to avoid bloating by duplicating strings (@tchaikov)

Implementing a limit is definitely a good idea, but I would be happy to have it be a simple assertion: that way we can just never merge anything that violates it (unit tests will fail on assertion)

**Comment by tchaikov** [comment]
i was changing `services`, `tags`, `see_also`, and `enum_allowed`, probably we should go further by changing `name`, `desc` and `long_desc` to `const char*` as well.

**Comment by liewegas** [comment]
That will make https://github.com/ceph/ceph/pull/18948/commits/3371a77bc353b5dfb271586964d10193479aacbb a bit more difficult, but manageable.  (We'll need to push the strings onto a list in md_config_t or something, similar to what we do with the Options themselves.)

File: src/common/options.h [hunk 1]
```diff
@@ -212,7 +213,11 @@ struct Option {
     return *this;
   }
   Option& set_description(const char* new_desc) {
-    desc = new_desc;
+    // only 100 single byte chars will be copied
+    char buf[max_desc_length];
+    // old school ...
+    memcpy(buf, new_desc, max_desc_length);
+    desc = buf;
```
Comment (liewegas) [inline src/common/options.h]: just switch this to assert(strlen(new_desc) < max_desc_length)

**Comment by jcsp** [comment]
@shinobu-x are you coming back to this?

**Comment by jcsp** [comment]
I'm closing this because it has been quiet for a long time, please feel free to reopen if anyone wants to finish this off.

**Pull request closed without merging**